### PR TITLE
Automation: Limit tests to not run on docs only changes

### PR DESCRIPTION
**Pull request by mkaz**
#23535  Description

Add filters for E2E, mobile, and performance tests to not run on documentation only changes, defined by *.md files in /docs/ and /packages/ directory

## How has this been tested?

Trying to figure out how to test. Ideally make a change to a markdown file and confirm that the E2E, Mobile, and Performance tests don't run, and don't block the PR.

## Types of changes

Github Actions workflow update adding exclude paths

### CODE REVIEW

**Comment by youknowriad** [comment]
The downside is the Github config to mark jobs as "Required". Now, we're forced to remove these tests from the required checks and just count on the "merger" to ensure these are passing before merging.

File: .github/workflows/end2end-test.yml
Comment (youknowriad) [inline .github/workflows/end2end-test.yml:7]: I don't think these are enough, I'd rather just keep the exclusions. changes to package.json could be impactful, to gutenberg.php and probably other things I'm missing.

**Comment by mkaz** [comment]
@youknowriad That was going to be my question, I wanted to test if they get skipped does that mean they are considered passed or not. Is there a way to return true?

File: .github/workflows/end2end-test.yml
Comment (mkaz) [reply .github/workflows/end2end-test.yml]: ok, I didn't know if exclusion only would work, I like that better 👍 
Updated to reflect that.

**Comment by mkaz** [comment]
@youknowriad We could remove the exclusion on the push to master check, but I suppose that doesn't really help, since its after the code got merged in.

**Comment by youknowriad** [comment]
i removed these tests from the "required" ones. I think it's essentially a Github issue, they shouuld allow an option that says "all checks" are required (and all would be dependent on the PR)

**Comment by mkaz** [comment]
@youknowriad I'm not sure it is necessary to remove those tests as being "required". They didn't run on this PR, so they might just be skipped altogether. Worth testing to see, harder to rely on a person having to remember to check.

**Comment by youknowriad** [comment]
>  I'm not sure it is necessary to remove those tests as being "required". 

It is necessary, otherwise you'll never be able to merge this PR (unless you're an admin). We had this issue already with other checks.

**Review by youknowriad — APPROVED** [review]

**Pull request merged**
